src/java/org/apache/cassandra/hints/HintsWriter.java

```diff
@@ -54,6 +56,9 @@ class HintsWriter implements AutoCloseable
 
     private volatile long lastSyncPosition = 0L;
 
+    @VisibleForTesting
+    AtomicLong totalHintsWritten = new AtomicLong();
```

Comment:
`totalHintsWritten` is only tracking new writes and on close we serialize it. What happens if there was a restart and statistics had some value? Shouldn't we set this to deserialized statistics value when creating `HintsWriter` ?

Reply:
This is good point. However, we always open a writer with CREATE_NEW option so we would never append to the file created before restart. However, it is worth storing statistics more frequently, like after each completed session.